### PR TITLE
Moving attribute table into own component

### LINKED ISSUE #1366: ViewComponent - Show - Attribute Table
Displays below the map and changes to look up attribute information when the map is clicked, if a WFS/WCS link is available.

Image

In the Stanford embed viewer, we do this as a custom leaflet overlay, so it's inside the viewer instead of outside it:

Image

### PR DESCRIPTION
**Pull request by hudajkhan**
Closes #1366 .
What this pull request does:

- Create attribute table component
- Use the component html erb file to display the attribute table container
- Passes in the value from the  "show_attribute_table" geoblacklight_helper method to determine if anything should be rendered
- Adds a simple test

### CODE REVIEW

File: app/components/geoblacklight/attribute_table_component.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+module Geoblacklight
+  ##
+  # A component for rendering attribute table
+  #
+  class AttributeTableComponent < ViewComponent::Base
+    attr_reader :show_attribute_table
```
Comment (dnoneill) [inline app/components/geoblacklight/attribute_table_component.rb:8]: Why do you have this if you aren't using it?

Comment (hudajkhan) [reply app/components/geoblacklight/attribute_table_component.rb:8]: You are correct : ) will remove

Comment (hudajkhan) [reply app/components/geoblacklight/attribute_table_component.rb]: Have removed

File: app/views/catalog/_show_default_attribute_table.html.erb [hunk 2]
```diff
@@ -1,19 +1 @@
...
-        <tbody class='attribute-table-body'>
-          <tr>
-            <td class='default-text' colspan='2'><em><%= t('geoblacklight.show.click_map') %></em></td>
-          </tr>
-        </tbody>
-      </table>
-    </div>
-  </div>
-<% end %>
+<%= render Geoblacklight::AttributeTableComponent.new(show_attribute_table:show_attribute_table?) %>
```
[the remark this answers is not in the document]
Comment (hudajkhan) [reply app/views/catalog/_show_default_attribute_table.html.erb]: Changed it as suggested above.

File: app/views/catalog/_show_default_attribute_table.html.erb [hunk 3]
```diff
@@ -1,19 +1 @@
-<% if show_attribute_table? %>
```
Comment (thatbudakguy) [inline app/views/catalog/_show_default_attribute_table.html.erb:1]: do we want to preserve this check? If so, I think all you need to do is add a `render?` to `AttributeTableComponent` that calls it

**Review by thatbudakguy — APPROVED** [review]

**Pull request merged**